app/assets/javascripts/two-pane.js

```diff
@@ -144,6 +144,11 @@ function right_pane_content(response){
   var contentId;
   if (handle_redirect(response)) return; //session expired redirect to login
 
+  // move flash messages from response to main window
+  $('.flash', response).each(function(index, item) {
```

Comment:
I see your point. I solved the problem on 1.14-stable branch. I did not expect, that the whole framework had been changed. But at least for 1.14 it solves the problem.
My critique with your solution is based on the fact, that `right_pane_content` loads the `div`-tag with class `content` or preferably class `two-pane-content` if available.
```
if (!$("#content", response).length){
    $('.two-pane-right').html(response);
    $('.two-pane-right form').prepend("<div class='fr close-button'><a class='two-pane-close' href='#'>&times;</a></div>");
    $('.form-actions a').addClass('two-pane-close');
    fix_multi_checkbox();
  } else {
    // response is not a form use the entire page
    if ($('#two_pane_content', response).length) {
      // replace only #two_pane_content if it's present in the response
      contentId = '#two_pane_content';
      two_pane_close();
    } else {
      contentId = '#content';
    }
    $(contentId).replaceWith($(contentId, response));
  }
```
I'm not sure whether this code path is used at all. But in case the AJAX response contains a `two-pane-content`, we would miss the notifications again.
Copying them to the main document, regardless where you find them seems the save way to me. Do you think, on the devel branch, we should copy everything inside `$("#notifications")` into the corresponding tag of the main document?